stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -2743,6 +2743,14 @@ end
 
 # Generic eigensystems
 eigvals(A::AbstractTriangular) = diag(A)
+function eigvecs(A::AbstractTriangular{T}) where {T<:BlasFloat}
+    Ac = eigencopy_oftype(A, T)
+    # Avoid a stack-overflow if the type doesn't change
+    if Ac isa typeof(A)
+        throw(ArgumentError(lazy"eigvecs type $(typeof(A)) not supported. Please submit a pull request."))
```

Comment:
`eigencopy_oftype(A)` carries out a conversion from `UpperTriangular(A::SMatrix)` to `UpperTriangular(::Matrix)`. Currently, this converts to whatever type is returned by `similar(A, T, size(A))`, which happens to be a `Matrix` in this case. Perhaps we should be explicit here in the conversion to a `Matrix`.

The check is added to ensure that we don't end up calling the same method if the `similar` call returns the same type.  This wouldn't be necessary if we explicitly convert to a `Matrix` because there exists a specialized `eigvecs` method for an `UpperTriangular(::Matrix)`.

Reply:
I've updated the implementation, and the flow should be simpler now